Minio/ApiEndpoints/IObjectOperations.cs

```diff
@@ -53,9 +53,10 @@ public interface IObjectOperations
         /// <param name="data">Stream of file to upload</param>
         /// <param name="size">Size of stream</param>
         /// <param name="contentType">Content type of the new object, null defaults to "application/octet-stream"</param>
+        /// <param name="metaData">Dictionary of metadata headers. Optional, defaults to null</param>
         /// <param name="cancellationToken">Optional cancellation token to cancel the operation</param>
 
-        Task PutObjectAsync(string bucketName, string objectName, Stream data, long size, string contentType = null, CancellationToken cancellationToken = default(CancellationToken));
+        Task PutObjectAsync(string bucketName, string objectName, Stream data, long size, string contentType = null, Dictionary<string, string> metadata=null, CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
This will break applications that were written based on our previous API. Can we salvage this by using optional arguments feature in C#? E.g,

```cs
Task PutObjectAsync(string bucketName, string objectName, Stream data, long size, string contentType = null, CancellationToken cancellationToken = default(CancellationToken), Dictionary<string, string> metadata=null);
```
Ref: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments#optional-arguments